pkg/secrets/provide_conjur_secrets.go

```diff
@@ -97,3 +96,55 @@ func RetryableSecretProvider(
 		return err
 	}
 }
+
+var TickerQuit = make(chan bool)
+var ProviderDone = make(chan bool)
+var SecretProviderError = make(chan error)
+func PeriodicSecretProvider(
+	secretRefreshInterval time.Duration,
+	mode string,
+	provideSecrets ProviderFunc,
+	) ProviderFunc {
+	return func() error {
+		err := provideSecrets()
+		if err != nil {
+			SecretProviderError <- err
+			return err
+		}
+		switch {
+		case mode != "sidecar":
+			SecretProviderError <- err
+			return err
+		case mode == "sidecar" && secretRefreshInterval > 0:
+			ticker := time.NewTicker(secretRefreshInterval)
+			go periodicSecretProvider(provideSecrets, ticker)
+		default:
+			// fall through
+		}
+		// Wait here
+		select {
+		case <-ProviderDone: // Todo change to SIGTERM or SIGINTERRUPT
+		}
+		if mode == "sidecar" && secretRefreshInterval > 0{
+			TickerQuit <- true
+			time.Sleep(1 * time.Second) // give a second for the go routine to shut down
+		}
+		return nil
+	}
+}
+
+func periodicSecretProvider(provideSecrets ProviderFunc, ticker *time.Ticker) {
```

Comment:
You can pass in the error channel (for output) and the quit channel (for input) to this function, so that you won't have to use global variables.

See: https://github.com/cyberark/secrets-provider-for-k8s/commit/d61c96c091f171be0aa04be5c75a6407a3e5f092#diff-c083eb56e1c331264db79fbb96d5fbb5ecfd8f082e9dde16e1a0e28f5f0f4230L136-R164